### DISCUSSION TITLE
passive aggressive behaviour

### DISCUSSION THREAD

**Discussion by timblechmann** [kovidgoyal/kitty#4661] [General]
@kovidgoyal, i'd like to apologise if i sounded passive aggressive in https://github.com/kovidgoyal/kitty/issues/4659.

while i appreciate your effort in improving the ecosystem, i doubt that this can be achieved by "kicking in its pants", but rather by providing a superior architecture and/or UX.

**Comment by page-down** [comment]
If all key combinations are disabled by default, as you suggest, how do I use them in the command line program?

Is this a superior UX?

**Comment by timblechmann** [comment]
> If all key combinations are disabled by default, as you suggest, how do I use them in the command line program?

afaict `konsole` for example uses an opt-in: in the "key bindings" list of the terminal profiles, it maps "some" shortcuts to escape codes and comes with a set of "reasonable defaults" ("reasonable" as in: as hardcore terminal user i never had to touch it).

**Comment by kovidgoyal** [comment]
By making key events available in default configuration, kitty encourages other programs to add support for them and allows users to use those key combinations by configuring other programs to use them, in the absence of explicit support. This is what will move this ecosystem forward. Not whatever you think konsole has been doing for decades with no effect.

And if you are actually in the habit of pressing random key combinations that you expect the software you use to be ignore, then I suggest you are unique. And yes you will need to add lots of lines in kitty.conf to disable all those random key combination you like to press for no reason. That's the price of progress. If on the other hand you find you often press one or two key combination accidentally, adding them to kitty.conf is not such a big burden.

In any case, I am done debating this. This has all been discussed ad nauseum when the keyboard protocol was first designed. I suggest you read the link I posted on your issue and the discussion of it that is also linked there.
